BasicSample/Assets/Shared/Scripts/AppRemotingSample.cs

```diff
@@ -48,15 +51,47 @@ private void Awake()
                 // into an XR experience and it's too late to connect remoting.
                 if (xrDisplaySubsystem.running)
                 {
-                    SetObjectActive(gameObject, false);
-                    SetObjectActive(immersiveUI, false);
-                    SetObjectActive(flatUI, false);
+                    DisableConnection2DUI();
                     return;
                 }
             }
 
-            SetObjectActive(immersiveUI, false);
-            SetObjectActive(flatUI, true);
+            ShowConnection2DUI();
+        }
+
+        private void Update()
+        {
+            var ip = remotingConfiguration.RemoteHostName ?? textInput.text;
+            var port = remotingConfiguration.RemotePort;
+
+            if (Remoting.AppRemoting.TryGetConnectionState(out m_connectionState, out m_disconnectReason))
```

Comment:
This feels like we might only want to enter the `if` if the state has changed. This might be why you were hitting an issue in `Remoting.AppRemoting.Disconnect()`, since it was being called every frame while the app is disconnected.

Reply:
Will refactor to avoid UI operation when state is not changed.

Reply:
In latest iteration, the UI change will only happen when connection state is changed.  Also refined the connection state message.